### PR TITLE
Improve release-cross tests

### PR DESCRIPTION
**Pull request by Ericson2314**
Ahead of https://github.com/NixOS/nixpkgs/pull/21388 or https://github.com/NixOS/nixpkgs/pull/21268, it would be nice to test cross-compiling better. This PR cleans up the existing tests, and adds some new ones. Those new ones fail, but that's OK: they won't hold up any channel, and they'll remind us to merge one of the two PRs above, which fixes the new tests.

@DavidEGrayson What do you think? I'd add you as a reviewer on the side panel, but It won't let me. If you like this I'll go ahead and merge it.

### CODE REVIEW

**Review by DavidEGrayson — APPROVED** [review]
The first commit was a little hard to review so I only glanced over it.  The rest looks good, though I did point out one nit.

File: pkgs/top-level/release-cross.nix [hunk 1]
```diff
@@ -20,39 +22,68 @@ let
...
+    # Converting to a string (drv path) before checking equality is probably a
+    # good idea lest there be some irrelevant pass-through debug attrs that
+    # cause false negatives.
+    testEqualOne = path: system: let
+      f = attrs: builtins.toString (lib.getAttrFromPath path (allPackages attrs));
+    in assert f { inherit system; } == f { inherit system crossSystem; }; true;
+
+    testEqual = path: systems: forAllSupportedSystems systems (testEqualOne path);
+
+    mapTestEqual = lib.mapAttrsRecursive (path: systems: testEqual path systems);
```
Comment (Ericson2314) [inline pkgs/top-level/release-cross.nix]: Oh hah! I few refactors in and I totally missed that worthless eta expansion. Thanks!

**Comment by Ericson2314** [comment]
http://hydra.nixos.org/eval/1317573 The good news is I didn't break any existing jobs, the bad news is that my `assert ...; true`s are not considered jobs.

**Comment by Ericson2314** [comment]
Those assertions should be failing, so I figured Hydra would have no idea whether there is a derivation behind them. Maybe it has some magic, but in any event I don't want the job to dissapear if the assert isn't thrown. I'll try something like `if .... then runCommand {} "true" else runCommand {} "false"`

**Comment by dezgeg** [comment]
FWIW they do show up in http://hydra.nixos.org/jobset/nixpkgs/cross-trunk#tabs-errors. But yes, you do need some sort of dummy derivations so they show up like the others.

**Comment by Ericson2314** [comment]
Oh, that's a good start. It's weird that that tab is there with all the evaluations, rather than per-evaluation.

**Comment by DavidEGrayson** [comment]
From this PR:

> Converting to a string (drv path) before checking equality is probably a
> good idea lest there be some irrelevant pass-through debug attrs that
> cause false negatives.

Equality comparison of derivations seems to be special and different from equality comparison of a normal set.  When you make the changes necessary to get those tests to pass, it proves that a derivation with a `crossDrv` inside it can be equal to a derivation without one.  So I bet that debugging attributes also don't matter.

**Pull request merged**